tests/integration/test_amax_amin.py

```diff
@@ -0,0 +1,237 @@
+        shape_list[axis] = 1
+        shape_true = tuple(shape_list)
+
+        res_np = np.random.randint(
+            1, 10, size=shape_true if keepdims else (5,) * (ndim - 1)
+        )
+        res_num = num.random.randint(
+            1, 10, size=shape_true if keepdims else (5,) * (ndim - 1)
+        )
+
+        func_np(in_np, axis=axis, out=res_np, keepdims=keepdims)
+        func_num(in_num, axis=axis, out=res_num, keepdims=keepdims)
+        assert np.array_equal(res_np, res_num)
+
+        val = 0.5
+        func_np(in_np, axis=axis, out=res_np, keepdims=keepdims, initial=val)
+        func_num(
+            in_num, axis=axis, out=res_num, keepdims=keepdims, initial=val
+        )
+        assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("func_name", FUNCS)
+def test_where(func_name):
+    # In Numpy, it pass
+    # In cuNumeric, it raises NotImplementedError
+    shape = (3, 4, 5)
+    in_np = np.random.random(shape)
+    in_num = num.array(in_np)
+    where_np = in_np > 0.5
+    where_num = num.array(where_np)
+
+    func_np = getattr(np, func_name)
+    func_num = getattr(num, func_name)
+
+    val = 0
+    assert np.array_equal(
+        func_np(in_np, initial=val, where=where_np),
+        func_num(in_num, initial=val, where=where_num),
+    )
+
+
+class TestAmaxAminErrors:
+    def setup_method(self):
+        size = (3, 4, 5)
+        self.arr_np = np.random.randint(-5, 5, size=size)
+        self.arr_num = num.array(self.arr_np)
+
+    @pytest.mark.parametrize("func_name", FUNCS)
+    def test_empty_array(self, func_name):
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            getattr(np, func_name)([])
+        with pytest.raises(expected_exc):
+            getattr(num, func_name)([])
+
+    @pytest.mark.parametrize(
+        "axis", (-4, 3), ids=lambda axis: f"(axis={axis})"
+    )
+    @pytest.mark.parametrize("func_name", FUNCS)
+    def test_axis_out_of_bound(self, func_name, axis):
+        expected_exc = ValueError
+        with pytest.raises(expected_exc):
+            getattr(np, func_name)(self.arr_np, axis=axis)
+        with pytest.raises(expected_exc):
+            getattr(num, func_name)(self.arr_num, axis=axis)
+
+    @pytest.mark.parametrize(
+        "axis_out_shape",
+        (
+            (None, (1,)),
+            (1, (3, 4)),
+        ),
+        ids=lambda axis_out_shape: f"(axis_out_shape={axis_out_shape})",
+    )
+    @pytest.mark.parametrize("func_name", FUNCS)
+    def test_out_invalid_shape(self, func_name, axis_out_shape):
+        axis, out_shape = axis_out_shape
+        expected_exc = ValueError
+        out_np = np.zeros(out_shape)
+        out_num = num.zeros(out_shape)
+
+        with pytest.raises(expected_exc):
+            getattr(np, func_name)(self.arr_np, axis=axis, out=out_np)
+        with pytest.raises(expected_exc):
+            getattr(num, func_name)(self.arr_num, axis=axis, out=out_num)
```

Comment:
I think all of these tests would be slightly clearer if you define `func_np` and `func_num` up front (like in the tests at the top of the PR) and then call those for the assertions

Reply:
Fixed. Thanks.